crates/miden-tx/src/host/mod.rs

```diff
@@ -519,27 +521,102 @@ where
+    ) -> Result<TransactionSummary, TransactionKernelError> {
+        let commitments = process.advice_provider().get_mapped_values(&msg).map_err(|err| {
+            TransactionKernelError::TransactionSummaryConstructionFailed(Box::new(err))
+        })?;
+
+        if commitments.len() != 16 {
+            return Err(TransactionKernelError::TransactionSummaryConstructionFailed(
+                "Expected 4 words for transaction summary commitments".into(),
+            ));
+        }
+
+        let salt = extract_word(commitments, 0);
+        let output_notes_commitment = extract_word(commitments, 4);
+        let input_notes_commitment = extract_word(commitments, 8);
+        let account_delta_commitment = extract_word(commitments, 12);
+
+        let account_delta = self.build_account_delta();
+        let input_notes = self.input_notes();
+        let output_notes_vec = self.build_output_notes();
+        let output_notes = OutputNotes::new(output_notes_vec).map_err(|err| {
+            TransactionKernelError::TransactionSummaryConstructionFailed(Box::new(err))
+        })?;
+
+        // Validate commitments
+        let actual_account_delta_commitment = account_delta.to_commitment();
+        if actual_account_delta_commitment != account_delta_commitment {
+            return Err(TransactionKernelError::TransactionSummaryCommitmentMismatch(
+                format!(
+                    "expected account delta commitment to be {actual_account_delta_commitment} but was {account_delta_commitment}"
+                )
+                .into(),
+            ));
+        }
+
+        let actual_input_notes_commitment = input_notes.commitment();
+        if actual_input_notes_commitment != input_notes_commitment {
+            return Err(TransactionKernelError::TransactionSummaryCommitmentMismatch(
+                format!(
+                    "expected input notes commitment to be {actual_input_notes_commitment} but was {input_notes_commitment}"
+                )
+                .into(),
+            ));
+        }
+
+        let actual_output_notes_commitment = output_notes.commitment();
+        if actual_output_notes_commitment != output_notes_commitment {
+            return Err(TransactionKernelError::TransactionSummaryCommitmentMismatch(
+                format!(
+                    "expected output notes commitment to be {actual_output_notes_commitment} but was {output_notes_commitment}"
+                )
+                .into(),
+            ));
+        }
+
+        Ok(TransactionSummary::new(account_delta, input_notes, output_notes, salt))
+    }
+
+    /// Aborts the transaction by building the
+    /// [`TransactionSummary`](miden_objects::transaction::TransactionSummary) based on elements on
+    /// the operand stack and advice map.
     ///
     /// Expected stack state:
     ///
     /// ```text
-    /// [SALT, OUTPUT_NOTES_COMMITMENT, INPUT_NOTES_COMMITMENT, ACCOUNT_DELTA_COMMITMENT]
+    /// [<word>, MESSAGE]
```

Comment:
It feels a bit weird that `MESSAGE` is the second word on the stack. Would have been a bit better if the expected state of the stack was just `[MESSAGE]`. I think this would require an extra `swapw` instruction - but maybe it's worth it?

Reply:
I agree it feels weird, but the way it currently is done matches the stack of `emit.AUTH_REQUEST`.
IMO it's much more important that `emit.AUTH_REQUEST` and `emit.UNAUTHORIZED` match, leading to less burden on the developer (who already has to juggle the two cases, which becomes quite evident in the multisig component @partylikeits1983 and I are working on).

Reply:
If we add a `swapw` before and after `emit.AUTH_REQUEST` then we could have `MESSAGE` at the top of the stack in both auth request and unauthorized cases, right? I think that would be worth it.

Reply:
Yes but we'd have to either swap back after the signature is on stack, or adapt the `verify` procedure to expect the message first

Reply:
After some more thinking and a chat with @PhilippGackstatter, I indeed think it's best to use the first word on the stack, done [here](https://github.com/0xMiden/miden-base/pull/1651/commits/623f424b2e220059ad72dca23506564812f0f5c0).

I also [adapted how we use `emit.AUTH_REQUEST` to match `emit.UNAUTHORIZED`](https://github.com/0xMiden/miden-base/pull/1651/commits/3cf62ce2d866b6919dd777fb4cf4f277fbcfcf43). 
With some [rearrangement of order of operations](https://github.com/0xMiden/miden-base/pull/1651/commits/7b452406c19c8dfc94a7956d24f5d49b55d6af32), we just need one extra `swapw` instead of two.